src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -804,6 +804,83 @@ public function testDumpNullAsTilde()
         $this->assertSame('{ foo: ~ }', $this->dumper->dump(['foo' => null], 0, 0, Yaml::DUMP_NULL_AS_TILDE));
     }
 
+    /**
+     * @dataProvider getNumericKeyData
+     */
+    public function testDumpInlineNumericKeyAsString(array $input, bool $inline, int $flags, string $expected)
+    {
+        $this->assertSame($expected, $this->dumper->dump($input, $inline ? 0 : 4, 0, $flags));
+    }
+
+    public function getNumericKeyData()
+    {
+        yield 'Int key with flag inline' => [
+            [200 => 'foo'],
+            true,
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            "{ '200': foo }",
+        ];
+
+        yield 'Int key without flag inline' => [
+            [200 => 'foo'],
+            true,
+            0,
+            '{ 200: foo }',
+        ];
+
+        $expected = <<<YAML
+'200': foo
+
+YAML;
```

Comment:
Let's use NOWDOC syntax and indent the block.

```suggestion
        $expected = <<<'YAML'
           '200': foo

           YAML;
```